pkg/tagger/collectors/kubernetes_metadata_mapper.go

```diff
@@ -64,8 +66,14 @@ func (c *KubeMetadataCollector) getTagInfos(pods []*kubelet.Pod) []*TagInfo {
 				log.Errorf("Could not fetch cluster level tags for the pod %s: %s", po.Metadata.Name, err.Error())
 				continue
 			}
-		} else {
+		} else if c.dcaClient.ClusterAgentVersion.Major >= 1 && c.dcaClient.ClusterAgentVersion.Minor >= 3 {
```

Comment:
not of a great fan of all the if/else logic for readability here, WDYT about isolating the part returning `metadataNames` in a function so that we can use early returns instead?

Reply:
Ok

Reply:
done